tests/ignite/metrics/gan/test_fid.py

```diff
@@ -0,0 +1,180 @@
+
+    mu1, sigma1 = train_samples.mean(axis=0), cov(train_samples, rowvar=False)
+    mu2, sigma2 = test_samples.mean(axis=0), cov(test_samples, rowvar=False)
+
+    sigma1 = torch.tensor(sigma1, dtype=torch.float64)
+    sigma2 = torch.tensor(sigma2, dtype=torch.float64)
+    assert pytest.approx(fid_score(mu1, mu2, sigma1, sigma2)) == pytorch_fid_score.calculate_frechet_distance(
+        mu1, sigma1, mu2, sigma2
+    )
+
+
+def test_compute_fid_from_features():
+    train_samples, test_samples = torch.rand(10, 10), torch.rand(10, 10)
+
+    fid_scorer = FID(num_features=10, feature_extractor=lambda x: x)
+    fid_scorer.update([train_samples[:5], test_samples[:5]])
+    fid_scorer.update([train_samples[5:], test_samples[5:]])
+
+    mu1, sigma1 = train_samples.mean(axis=0), cov(train_samples, rowvar=False)
+    mu2, sigma2 = test_samples.mean(axis=0), cov(test_samples, rowvar=False)
+
+    assert pytest.approx(pytorch_fid_score.calculate_frechet_distance(mu1, sigma1, mu2, sigma2)) == fid_scorer.compute()
+
+
+def test_wrong_inputs():
+    with pytest.raises(ValueError, match=r"num of features must be greater to zero"):
+        FID(num_features=-1, feature_extractor=lambda x: x)
+    with pytest.raises(ValueError, match=r"Features must be a tensor of dim 2 \(got: 1\)"):
+        FID(num_features=1, feature_extractor=lambda x: x).update(torch.Tensor([[], []]))
+    with pytest.raises(ValueError, match=r"Batch size should be greater than one \(got: 0\)"):
+        FID(num_features=1, feature_extractor=lambda x: x).update(torch.rand(2, 0, 0))
+    with pytest.raises(ValueError, match=r"Feature size should be greater than one \(got: 0\)"):
+        FID(num_features=1, feature_extractor=lambda x: x).update(torch.rand(2, 2, 0))
+
+
+def test_statistics():
+    train_samples, test_samples = torch.rand(10, 10), torch.rand(10, 10)
+    fid_scorer = FID(num_features=10, feature_extractor=lambda x: x)
+    fid_scorer.update([train_samples[:5], test_samples[:5]])
+    fid_scorer.update([train_samples[5:], test_samples[5:]])
+
+    mu1, sigma1 = train_samples.mean(axis=0), torch.tensor(cov(train_samples, rowvar=False))
+    mu2, sigma2 = test_samples.mean(axis=0), torch.tensor(cov(test_samples, rowvar=False))
+
+    fid_mu1 = fid_scorer._train_total / fid_scorer._num_examples
+    fid_sigma1 = fid_scorer.get_covariance(fid_scorer._train_sigma, fid_scorer._train_total)
+
+    fid_mu2 = fid_scorer._test_total / fid_scorer._num_examples
+    fid_sigma2 = fid_scorer.get_covariance(fid_scorer._test_sigma, fid_scorer._test_total)
+
+    assert torch.isclose(mu1.double(), fid_mu1).all()
+    for cov1, cov2 in zip(sigma1, fid_sigma1):
+        assert torch.isclose(cov1.double(), cov2, rtol=1e-04, atol=1e-04).all()
+    assert torch.isclose(mu2.double(), fid_mu2).all()
+    for cov1, cov2 in zip(sigma2, fid_sigma2):
+        assert torch.isclose(cov1.double(), cov2, rtol=1e-04, atol=1e-04).all()
+
+
+# def test_inception_extractor_wrong_inputs():
```

Comment:
We have to discuss this. Downloading inception weights is a cause of problems... We could let the user gives the model, or introduce a parameter to disable pretraining. 

@vfdev-5 Shall you have a recommendation on this issue ?